src/main/scala/org/constellation/p2p/Cluster.scala

```diff
@@ -463,18 +461,34 @@ class Cluster[F[_]: Concurrent: Logger: Timer: ContextShift](ipManager: IPManage
       "cluster_leave"
     )
 
-  def broadcastNodeState(): F[Unit] =
+  private def broadcastNodeState(nodeState: NodeState): F[Unit] =
     logThread(
-      getNodeState.flatMap { nodeState =>
-        broadcast(_.postNonBlockingUnitF("status", SetNodeStatus(dao.id, nodeState))(C))
-      }.flatTap {
+      broadcast(_.postNonBlockingUnitF("status", SetNodeStatus(dao.id, nodeState))(C)).flatTap {
         _.filter(_._2.isLeft).toList.traverse {
           case (id, e) => Logger[F].warn(s"Unable to propagate status to node ID: $id")
         }
       }.void,
       "cluster_broadcastNodeState"
     )
 
+//  def compareAndSet(expected: NodeState, newState: NodeState, skipBroadcast: Boolean = false): F[SetStateResult] =
```

Comment:
?

Reply:
@marcinwadon do we need perform snapshots during leaving process ? e.g wait for 2 snapshots ?